bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/relay/RelayHandler.java

```diff
@@ -0,0 +1,320 @@
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    private void configurePowerSwitchModeChannels() {
+        List<String> channelsToBePresent = List.of(CHANNEL_POWER_SWITCH);
+        List<String> channelsToBeAbsent = List.of(CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH,
+                CHANNEL_INSTANT_OF_LAST_IMPULSE);
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    /**
+     * Re-configures the channels of the associated thing, if applicable.
+     * 
+     * @param channelsToBePresent channels to be added, if not present already
+     * @param channelsToBeAbsent channels to be removed, if present
+     */
+    private void configureChannels(List<String> channelsToBePresent, List<String> channelsToBeAbsent) {
+        List<String> channelsToAdd = channelsToBePresent.stream().filter(c -> getThing().getChannel(c) == null)
+                .toList();
+        List<Channel> channelsToRemove = channelsToBeAbsent.stream().map(c -> getThing().getChannel(c))
+                .filter(Objects::nonNull).map(Objects::requireNonNull).toList();
+
+        if (channelsToAdd.isEmpty() && channelsToRemove.isEmpty()) {
+            return;
+        }
+
+        ThingBuilder thingBuilder = editThing();
+        if (!channelsToAdd.isEmpty()) {
+            addChannels(channelsToAdd, thingBuilder);
+        }
+        if (!channelsToRemove.isEmpty()) {
+            thingBuilder.withoutChannels(channelsToRemove);
+        }
+
+        updateThing(thingBuilder.build());
+    }
+
+    private void addChannels(List<String> channelsToAdd, ThingBuilder thingBuilder) {
+        for (String channelToAdd : channelsToAdd) {
+            Channel channel = createChannel(channelToAdd);
+            thingBuilder.withChannel(channel);
+        }
+    }
+
+    private Channel createChannel(String channelId) {
+        ChannelUID channelUID = new ChannelUID(getThing().getUID(), channelId);
+        ChannelTypeUID channelTypeUID = getChannelTypeUID(channelId);
+        @Nullable
+        String itemType = getItemType(channelId);
+        return ChannelBuilder.create(channelUID, itemType).withType(channelTypeUID).build();
+    }
+
+    private ChannelTypeUID getChannelTypeUID(String channelId) {
+        switch (channelId) {
+            case CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH, CHANNEL_INSTANT_OF_LAST_IMPULSE:
+                return new ChannelTypeUID(BINDING_ID, channelId);
+            case CHANNEL_POWER_SWITCH:
+                return DefaultSystemChannelTypeProvider.SYSTEM_CHANNEL_TYPE_UID_POWER;
+            default:
+                throw new UnsupportedOperationException(
+                        "Cannot determine channel type UID to create channel " + channelId + " dynamically.");
+
+        }
+    }
+
+    private @Nullable String getItemType(String channelId) {
+        switch (channelId) {
+            case CHANNEL_POWER_SWITCH, CHANNEL_IMPULSE_SWITCH:
+                return "Switch";
```

Comment:
@jlaur are there any constants available for the item types?

Reply:
Doesn’t look like it.

Reply:
ok, thanks for looking into it anyways @lsiepel :+1:

Reply:
https://github.com/openhab/openhab-core/blob/main/bundles%2Forg.openhab.core%2Fsrc%2Fmain%2Fjava%2Forg%2Fopenhab%2Fcore%2Flibrary%2FCoreItemFactory.java#L59

Reply:
Thanks 🙂 Is there also a constant for `Time` as in `Number:Time`?

Reply:
I don't know for sure, but I don't think so. You could try searching the core repo for "Time" (including the quotes).

Reply:
I did not find a constant for `Time`, but used all available item type constants in the latest commit.